src/Symfony/Component/String/AbstractString.php

```diff
@@ -685,11 +685,34 @@ public function wordwrap(int $width = 75, string $break = "\n", bool $cut = fals
         return $str;
     }
 
+    public function __serialize(): array
+    {
+        $data = [];
+        foreach ($this->__sleep() as $key) {
+            $data[$key] = $this->$key;
+        }
+
+        return $data;
+    }
+
+    public function __unserialize(array $data): void
+    {
+        foreach ($data as $key => $value) {
+            $this->$key = $value;
+        }
+
+        $this->__wakeup();
+    }
+
     public function __sleep(): array
     {
         return ['string'];
     }
 
+    public function __wakeup(): void
```

Comment:
We have to add the method so that __unserialize can call it, in case a child class defines it.
We could do a method_exists check, but that's be adding more boilerplate.